### ISSUE TITLE
Need clear installation documentation

### ISSUE THREAD

**Issue by dbbc96** [aws/karpenter#4722]
### Description

So i looking at your documentation on installation (specifically on EKS in AWS), i see some gaps in what i feel need to addressed.  Some of the links pointing to AWS Blueprints are either broken or provide little help.  What i would like to see is clearly the steps on installing Karpenter on an existing cluster.  For example step 1) tagging the resources (and go into that detail)  2) Setup IAM roles and policies (go into this as there is no documentation that i saw that talks about needing SQS. however it looks to be needed or you will get errors from the logs) 3) install CRDs and helm stuff along with authmap stuff.  

Most of the steps i have seen are for users when migrating from Cluster AutoScaler but if someone isn't doing that then there no reason for a user to click on that.  The AWS Blueprints do have a lot of the terraform however jumping between the "addon" and "addons" is a bit much and noisy and has other stuff that isn't needed to just get Karpenter installed.  
The cloudformation mentioned in your docs also appears to setup more than needed if the infrastructure is setup and just needs Karpenter.

**Comment by njtran** [comment]
https://github.com/aws/karpenter/pull/4394 should make this better.

**Comment by dbbc96** [comment]
@njtran i see this is talking about CloudFormation.  Not sure if this will only focus on CF setup.  If this is the case, its not exactly what i was looking for.  Please let me know know.  All that i was inquiring about is just the main setup required for Karpenter with the assumption that someone already has a cluster setup but without anything else.  So just looking for the steps for each component that Karpenter needs to work correctly.  We use Terraform however not expecting the docs to list terraform.  It should be the requirements without any assumptions on the method that someone wishes to implement.

**Comment by njtran** [comment]
From what I understand, the CFN PR will make it easier to understand all the resources you need to create for the Karpenter install, which most users will require. The other bit about tagging should be covered with https://karpenter.sh/docs/getting-started/migrating-from-cas/. @chrisnegus can chime in here, but I think there's an overarching effort to make this better. Which part of the migrating-from-cas guide is not sufficient for you? Can you share which lines/sections in particular? 

If you're willing to help contribute to the docs I can work with you on this.

**Comment by dbbc96** [comment]
my suggestion would be a central place for all the "pre-reqs".  the migrating from cas section, is fine for someone who would be wanting to migrate.  For someone who never installed cas, a regular user would never go there.  What you could do instead is have a page for all the pre-reqs like iam, tagging, sqs, authmap, etc. and then in the cas guide, they reference this new page.  Cause the pre-reqs would be required for any user who is new or migrating from cas

**Comment by njtran** [comment]
Got it. @chrisnegus should be working on this. If not the PR that I linked, he should be able to eventually get to it. @chrisnegus is this the best place to track this effort?

**Comment by chrisnegus** [comment]
@njtran Yes, let's track this effort here. I think much of what is described here is covered in the new [CloudFormation](https://karpenter.sh/preview/reference/cloudformation/) page. But I'll go through and see if any of the other issues brought up still need to be added to the docs.

**Comment by dbbc96** [comment]
i will wait till you all have finished but one other comment is tool make documentation tool agnostic as not everyone uses cloudformation.

**Comment by chrisnegus** [comment]
The CloudFormation page doesn't cover everything that @dbbc96 is looking for. I had started work on such a guide (essentially, migrating to Karpenter from any existing cluster). I'll try to cover these requests when I pick up work on that guide again. For now, I added a link from the getting started guide to the CloudFormation page in [PR #4997](https://github.com/aws/karpenter/pull/4997).

**Comment by e-koma** [comment]
I agree with @dbbc96 

the migration guide, `Migrating from Cluster Autoscaler`,  includes instructions on how to install CRDs, but these instructions are absent from the installation guide. This means that people who only look at the `Getting Started with Karpenter` section will not have enough information.


The migration guide details the CRD installation commands as follows:

```
kubectl create -f \
    "https://raw.githubusercontent.com/aws/karpenter-provider-aws/v${KARPENTER_VERSION}/pkg/apis/crds/karpenter.sh_nodepools.yaml"
kubectl create -f \
    "https://raw.githubusercontent.com/aws/karpenter-provider-aws/v${KARPENTER_VERSION}/pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml"
kubectl create -f \
    "https://raw.githubusercontent.com/aws/karpenter-provider-aws/v${KARPENTER_VERSION}/pkg/apis/crds/karpenter.sh_nodeclaims.yaml"
```

However, these are not mentioned in the installation guide. 
Correcting the CloudFormation is not sufficient. the CRDs remain uninstalled.

`Getting Started with Karpenter` should include missing information, such as applying CRDs.

**Comment by SebastianScherer88** [comment]
@dbbc96 i have tweaked [the official terraform karpenter example](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/karpenter) (removed the fargate hosting of the controller and addressed [the service linked role issue](https://github.com/terraform-aws-modules/terraform-aws-eks/issues/2565)) in this [directory of my personal repo](https://github.com/SebastianScherer88/ml-ops-demo/tree/main/eks-karpenter).

if you want to really drill down into the resource level dependencies (as you thought, the SQS is a requirement/dependency), you can dig into[ the official terraform karpenter module](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/modules/karpenter) that's being used in the example.

unfortunately these really useful resources havent made their way into the karpenter docs yet, it seems.

hope that helps!

**Comment by billnbell3** [comment]
For AWS Fargate it would be good to run karpenter namespace on Fargate nodes for higher reliability.

I tried doing this and it was giving me multiple errors in split configuration. Karpenter pods on Fargate node and my workload on Karpenter nodes. I keep getting API errors and DNS issues.

Maybe good documentation can help ?

**Comment by dbbc96** [comment]
@billnbell3 its been a bit since i've setup my lab environment but the last did Karpenter, i recall i was able to get Karpenter on Fargate and have the rest work on ec2 worker nodes.  I would have to double check to see if that is still the case.

**Comment by billnbell3** [comment]
It would be good if we had a walk thru for Terraform on how to setup karpenter on fargate. I jsut tried it again, and it failed, and never comes up. Some kind of chicken and egg problem. We need to have managed nodes up to get karpenter to come up, and then maybe we can delete the managed nodes. Or maybe it is better to run the add ons using fargate, and karpenter can manage the pods with nodes?

I think a simple explanation of best practices would be best for Karpenter and Fargate. And then how do you run system stuff in fargate and rest in Karpenter. And a working example of how to do it in latest version from scratch ?

**Comment by dbbc96** [comment]
So I wanted to come back to this.  So again I am asking for just a simple doc that shows what is required for managed nodes vs fargate.  I just went through terraform module and built out 3 different cases where 1). EKS Auto Mode (no need to do Karpenter) 2. Managed Nodes (Karpenter on ec2), and 3) Managed Nodes with Karpenter on Fargate.  I don't do eksctl or Cloudformation and so having instructions (or screenshots) of the IAM policy and roles for Karpenter would be ideal.  It would allow users to build out using their choice of tools.  Because now there is Pod Identity (not on Fargate) it's good to show the differences.  to my understanding also that SQS is optional unless doing spot instances.
